### PR TITLE
I884 publish local with header

### PR DESCRIPTION
**Pull request by laneb**
Addressing #884.

### CODE REVIEW

File: src/main/scala/org/tresamigos/smv/SmvReportIO.scala [hunk 1]
```diff
@@ -30,16 +30,11 @@ private[smv] object SmvReportIO {
...
-
-    // Save report RDD to a temparory file on HDFS
-    val tmpHdfsFile = "/tmp/smv_tmp_" + filename
-    SmvHDFS.deleteFile(tmpHdfsFile)
-    report.saveAsTextFile(tmpHdfsFile)
-
-    // copy merge the HDFS output to a local output
-    SmvHDFS.copyMerge(tmpHdfsFile, path)
-    SmvHDFS.deleteFile(tmpHdfsFile)
+    val pw = new PrintWriter(outFile)
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/SmvReportIO.scala]: Where is the header being written?

File: src/main/scala/org/tresamigos/smv/SmvReportIO.scala [hunk 2]
```diff
@@ -30,16 +30,11 @@ private[smv] object SmvReportIO {
...
+    // According to Spark API doc, RDD.toLocalIterator will consume no more
+    // memory than what is required for the largest partition
+    report.toLocalIterator foreach (r => pw.write(r.toString + "\n"))
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/SmvReportIO.scala]: I'm assuming the type of `r` is `Row`.  Is `r.toString` taking care of quoting special characters?

File: src/main/scala/org/tresamigos/smv/SmvReportIO.scala (hunk 1 above)
Comment (laneb) [inline src/main/scala/org/tresamigos/smv/SmvReportIO.scala]: The header is the first entry of the RDD.

File: src/main/scala/org/tresamigos/smv/SmvReportIO.scala (hunk 2 above)
Comment (laneb) [inline src/main/scala/org/tresamigos/smv/SmvReportIO.scala]: Actually, the type of `r` is `String` and does not need the additional `toString`. Will fix.

**Comment by laneb** [comment]
Just noticed a behavior change on this branch. Hadoop was automatically creating directories for us if we tried to copy-merge into a directory that didn't exist. With the current implementation, there will be an error if the directory doesn't exist. Need to fix that.

**Comment by laneb** [comment]
Actually, per @AliTajeldin the behavior should not be to create the directory automatically. We will verify that the directory exists and throw an error otherwise.

File: src/main/scala/org/tresamigos/smv/SmvReportIO.scala [hunk 3]
```diff
@@ -17,24 +17,34 @@ package org.tresamigos.smv
 import org.apache.spark.rdd.RDD
 import java.io.{PrintWriter, File}
 
+private[smv] class ReportWriter(path: String) {
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/SmvReportIO.scala]: guess what's missing from this class definition :smile:

Comment (laneb) [inline src/main/scala/org/tresamigos/smv/SmvReportIO.scala]: -_-

**Pull request merged**